### ISSUE TITLE
Python controller module continuous reloading doesn't work

### ISSUE THREAD

**Issue by lordloki** [UPBGE/upbge#1617]
Currently, the option for continuous reloading of python module doesn't work (first time works but subsequent times don't work)

![image](https://user-images.githubusercontent.com/1015605/147295011-138a4761-1cc1-413f-b3b9-0ae80fd83073.png)

The issue is because in the SCA_PythonController.cpp under Import function:
```
  // Import the module and print an error if it's not found
  PyObject *mod = PyImport_ImportModule(mod_path.c_str());

  if (mod == nullptr) {
    ErrorPrint("Python module can't be imported");
    return false;
  }

  if (m_debug)
    mod = PyImport_ReloadModule(mod);

  if (mod == nullptr) {
    ErrorPrint("Python module can't be reloaded");
    return false;
  }
```
The function PyImport_ReloadModule return nullptr.

This issue is due to some commit between 2021/01/10 (https://github.com/UPBGE/upbge/commits/master?after=51cd0bfe33ff2ee1f85ff69ce46d6732c9ee9352+9500&branch=master) and 2021/01/14 (https://github.com/UPBGE/upbge/commits/master?before=51cd0bfe33ff2ee1f85ff69ce46d6732c9ee9352+9396&branch=master).

Additionally, to comment that it works when you are in debug mode.

Here is the test file: [001_reloadme.zip](https://github.com/UPBGE/upbge/files/7771590/001_reloadme.zip)

**Comment by youle31** [comment]
from my investigations it comes from this commit: 62bff15377510ce05543afb40526d430a21697f1

**Comment by lordloki** [comment]
Using blenderplayer it works well

**Comment by lordloki** [comment]
It is very strange. I have a build from 11/01/2021 hash: 5a106ab39 that works perfectly.
And I just compile this exact hash right now (doing a checkout of old libraries, etc) and the build doesn't work.
Maybe could something related to the MSVC compiler? It is strange.

**Comment by youle31** [comment]
weird...

**Comment by HolandonHdunto** [comment]
Speaking of python and module, I can also take a look at pothon language and a clone of python language, but it is faster and can run on Windows, Linux, MacOS, Android, IOs, HTML5, Xbox, Playstation, etc. I think from pothon, programming with a python-like language, you can convert and export as c++, c, java and so on.
 https://github.com/HashLDash/Photon

**Comment by youle31** [comment]
At first glance this patch is working, however idk if it can break other parts of the code:

[p.zip](https://github.com/UPBGE/upbge/files/7879712/p.zip)

**Comment by lordloki** [comment]
Fixed in master. Thanks to @youle31

**Comment by 3DSki** [comment]
I'm trying this same setup, expecting it to be a simple demo of something that works, but doesn't. First, I had mixed behaviors with the book's *.blend and I questioned the version of Blender it was created with.  If I didn't set continuous the cube would spin faster than expected. If I changed the file and restarted the game, nothing would change, not even if i change the axis of rotation.

I updated the *.blend to use the default scene and then added the same script.reload_me module usage. In this scenario, the cube is not effected at all, regardless if continuous reload is set or not. I've attached a ZIP of the files.
[reload.zip](https://github.com/UPBGE/upbge/files/12284052/reload.zip)

**Comment by youle31** [comment]
I tested in 0.36 and in 0.4alpha and it was working as expected like in this video:

[reload_video_demo.zip](https://github.com/UPBGE/upbge/files/12286899/reload_video_demo.zip)

**Comment by 3DSki** [comment]
Thanks for your time. I'll take a look again. I'm new and haven't learned
how to look deeper at the processing to see the moving parts and understand
all that settings that might affect a particular function..

**Comment by 3DSki** [comment]
I repeated the process and made sure I started blender from the console so
I could see what might be logged. It's failing to find and load the
script.py module, even though it's in the same directory as the blend file.
What determines where Blender will look for a module? The UI doesn't
provide a file browser to pick, so one would assume that it would always be
the location of the blend file. Also, I can't open my blend file through
the file explorer, because I have other versions of Blender installed and
it would be associated with the standard Blendar 3.6 instead of UPBGE.
Things work as expected if I perform the same thing with the Range version
(script.py is found at the blend file location).
